src/genie_python/genie_blockserver.py

```diff
@@ -27,38 +35,38 @@ def wrapper(*args, **kwargs):
 
 
 class BlockServer(object):
-    def __init__(self, api):
-        self.api = api
+    def __init__(self, api: API) -> None:
+        self.api: API = api
 
-    def _get_pv_value(self, pv, as_string=False):
+    def _get_pv_value(self, pv: str, as_string: bool = False) -> PVValue:
         """Just a convenient wrapper for calling the api's get_pv_value method"""
         return self.api.get_pv_value(self.api.prefix_pv_name(pv), as_string)
 
-    def _set_pv_value(self, pv, value, wait=False):
+    def _set_pv_value(self, pv: str, value: PVValue | bytes, wait: bool = False) -> None:
         """Just a convenient wrapper for calling the api's set_pv_value method"""
         return self.api.set_pv_value(self.api.prefix_pv_name(pv), value, wait)
 
     @_blockserver_retry
-    def get_sample_par_names(self):
+    def get_sample_par_names(self) -> str:
         """Get the current sample parameter names as a list."""
         # Get the names from the blockserver
         raw = self._get_pv_value(BLOCK_SERVER_PREFIX + "SAMPLE_PARS", True)
         return dehex_decompress_and_dejson(raw)
 
     @_blockserver_retry
-    def get_beamline_par_names(self):
+    def get_beamline_par_names(self) -> str:
         """Get the current beamline parameter names as a list."""
         # Get the names from the blockserver
         raw = self._get_pv_value(BLOCK_SERVER_PREFIX + "BEAMLINE_PARS", True)
         return dehex_decompress_and_dejson(raw)
 
     @_blockserver_retry
-    def get_runcontrol_settings(self):
+    def get_runcontrol_settings(self) -> str:
         """Get the current run-control settings."""
         raw = self._get_pv_value(BLOCK_SERVER_PREFIX + "GET_RC_PARS", True)
         return dehex_decompress_and_dejson(raw)
```

Comment:
I think with these `dejson` ones the type hint needs to be what the JSON parses to... which you can either type hint "properly" as `dict[dict[<whatever the scructure is>]]`, or more pragmatically, as `Any`

Reply:
is it sensible to do this on the ones that claim to just be a list of names since that isnt programmatically guaranteed?

Reply:
If you "know" that the PV should contain data in a particular format, you could assert that and raise an error if not... But I'm ok with just the cheap get-out `Any` type here tbh as that'll be tedious.